### DISCUSSION TITLE
base/Element.php class, indexHtml() function - add sourceKey variable to twig file context

### DISCUSSION THREAD

**Discussion by piotrpog** [craftcms/cms#10521] [Ideas] [answered]
Hi,
i am creating plugin that adds filter widgets to element indexes lists. Currently i am using JS to inject widgets markup, but i want to do that on backend side of things, using `EVENT_AFTER_RENDER_TEMPLATE` event. 

This event allows me to modify HTML code of rendered template (for example using php domdocument), i also have access to Twig variables. Hovewer to inject proper filters, i need to have access to `sourceKey` variable that is used in `indexHtml()` function used in `base/Element.php` class - but it is not passed to rendered `_elements/tableview/container` template.

Would it be possible to pass this variable to template file?

**Comment by piotrpog** [comment, marked as answer]
Well it just occured to me that i can access POST request that contains `source` param - so i guess my request is unnecessary.

**Reply by brandonkelly** [reply]
Yeah that works… and probably safer than us starting to pass a template variable that doesn’t seem to be needed.
